### DISCUSSION TITLE
Installing DITA open toolkit and working through Learning DITA courses

### DISCUSSION THREAD

**Discussion by TechWritingFitz** [dita-ot/dita-ot#4075] [Q&A]
Hey everyone, 

I'm a novice technical writer, who is learning DITA through the many brilliant open-source resources that DITA Open Toolkit has made available to us. 

It might be that, through my newness to the field, I am missing the link between sample exercises in the Learning DITA courses, the Java Environment and DITA 4.0, but the process of opening these exercises using DITA 4.0 seemed really clunky as described in the Toolkit itself. I tried opening via the command line on my laptop having downloaded both JAVA and DITA, but didn't get any joy, so in the end I created a new Github repository and imported them here, where they've opened fine. I now need to think about how I work through the sample exercises in Github, maybe via syncing with Stackedit? 

Are there any other newbies out there who have had similar issues, and am I responding to these sorts of challenges correctly? Furthermore, is there an argument for simplifying the instructions for beginners? 

Cheers!

Kieran

**Comment by chrispy-snps** [comment]
@TechWritingFitz - welcome to the DITA community!

DITA can be simple or complex depending on your needs, and information is distributed in many places - some newer, some older. It can definitely be challenging for beginners to find their bearings.

* What operating system are you using?
* You mentioned you downloaded Java - I assume that means you downloaded _and_ installed it?
* For the DITA-OT, did you follow the installation and validation instructions [here](https://www.dita-ot.org/dev/topics/installing-client.html)?

I haven't tried the Learning DITA courses, but if Java and the DITA-OT are both installed in the usual way, you should be able to run a transformation simply by running the `dita` or `dita.bat` command.

If you used the installation procedures above and it isn't working, what error message did you get?

**Reply by TechWritingFitz** [reply]
Hi Chris, 

Thanks for your reply, I've found it really encouraging. I'll respond to your queries in order:

- I'm on 2013 Mac, running OSX Big Sur. 
- Yep, I downloaded the .pkg file for installing the Java SE Development Kit (v.19) and installed it. I just repeated this process to make sure I have it installed correctly.
- I actually think I might be tripping up on this step, particularly as regards extracting the DITA package from its download and installing it into a file directory. I thought this would be as simple as saving it to my desktop, but when I click through to the "bin" folder and try to open the dita file I get a security message, which reads, "dita cannot be opened because it is from an unidentified developer. MacOS cannot verify that this app is free from malware". There doesn't seem to be any way of over-riding this; I just have an "OK" button to click. 

Thanks for the assistance!

Best, 

Kieran

**Reply by chrispy-snps** [reply]
I'm not familiar with MacOS, so my advice might not be complete.

You will work with the DITA-OT in a terminal window, so you will need some basic experience with that.

In a terminal window, does the `java -version` command show you the expected v19 version? You could also verify that the `which java` command shows the expected installation location.

The DITA-OT .zip file should be extracted to somewhere; it cannot be run from inside the .zip file (assuming MacOS allows you to "look inside" a .zip file like Windows does). In a terminal window, you should be able to change your current directory to where the DITA-OT directory is, then check the version like this:

```
% cd <path_to_DITAOT>
% ls bin
ant  ant.bat  dita  dita.bat
% bin/dita -version
DITA-OT version 4.0.1
%
```

If I do an internet search for the "cannot be opened because it is from an unidentified developer" message, I see some suggestions on how to resolve it, but I have no way to investigate further.

**Reply by TechWritingFitz** [reply]
Hi Chris, 

Yes, having run those two commands, Java is installed correctly with v19. 

I've extracted the DITA-OT to my Dropbox directly and I'm getting some success in the Terminal line now, which is great! Thanks to you and @raducoravu for the assistance so far. I'm still stalling a little on generating the HTML output, but I think I just need to spend more time makings trials and errors to figure it out. 

Thanks a million, 

Kieran

**Comment by raducoravu** [comment]
@TechWritingFitz the DITA Open Toolkit is not an application, does not have a graphical user's interface, it needs to be run from the command line:
https://www.dita-ot.org/dev/topics/build-using-dita-command.html
So open a Terminal, "cd" to the DITA OT folder and then "cd" to the "bin" folder and then run the command which generates for example the HTML5 output:

       cd Downloads/dita-ot-3.7.4/
       cd bin/
       sh dita -i /Users/username/path/to/sample.ditamap -f html5 -o /Users/username/path/to/output/folder

About errors like this "dita cannot be opened because it is from an unidentified developer. Every downloaded archive or application on MacOS has a quarantine flag and when running executable files from the Mac OS Finder, the operating system checks if the ZIP was signed before running the executable. You can remove the quarantine flag for a folder by running in the Terminal a command like:

     xattr -dr com.apple.quarantine Downloads/dita-ot-3.7.4/

but as I said the DITA OT is a command line tool so you need to start the publishing from the Terminal.
